### PR TITLE
Add additional milisecond in test_search_experiments_filter_by_time_attribute

### PR DESCRIPTION
**Pull request by nojaf**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/nojaf/mlflow/pull/14025?quickstart=1)

#### Install mlflow from this PR

```
# Use `%sh` to run this command on Databricks
OPTIONS=$(if pip freeze | grep -q 'mlflow @ git+https://github.com/mlflow/mlflow.git'; then echo '--force-reinstall --no-deps'; fi)
pip install $OPTIONS git+https://github.com/mlflow/mlflow.git@refs/pull/14025/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 14025
```

</p>
</details>

### Related Issues/PRs

Occasionally, `tests/store/tracking/test_sqlalchemy_store.py::test_search_experiments_filter_by_time_attribute` fails for us in mlflow-go.

I noticed that on Mac, the assert in

```py
    now = get_current_time_millis()
    experiments = store.search_experiments(filter_string=f"creation_time >= {now}")
    assert experiments == []
```

can fail if the second experiment is created in the exact same milisecond as the `now`.
This can happen for me on my Mac locally.

### What changes are proposed in this pull request?

We just add an additional millisecond to avoid this problem.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 1]
```diff
@@ -474,24 +487,33 @@ def test_search_experiments_filter_by_time_attribute(store: SqlAlchemyStore):
...
-    experiments = store.search_experiments(filter_string=f"creation_time < {time_before_create2}")
+    experiments = store.search_experiments(
+        filter_string=f"creation_time < {time_before_create2}"
+    )
     assert [e.experiment_id for e in experiments] == [
         exp_id1,
         store.DEFAULT_EXPERIMENT_ID,
     ]
 
+    time.sleep(0.001)
```
Comment (nojaf) [inline tests/store/tracking/test_sqlalchemy_store.py:516]: This is the actual change to ensure `now` below is not the same value as the `exp_id2` create_time.

**Review by harupy — APPROVED** [review]
LGTM

File: tests/store/tracking/test_sqlalchemy_store.py [hunk 2]
```diff
@@ -1306,10 +1307,12 @@ def test_set_tag(store: SqlAlchemyStore, monkeypatch):
     # test setting tags that are too long fails.
     monkeypatch.setenv("MLFLOW_TRUNCATE_LONG_VALUES", "false")
     with pytest.raises(
-        MlflowException, match=f"exceeds the maximum length of {MAX_TAG_VAL_LENGTH} characters"
+        MlflowException,
+        match=f"exceeds the maximum length of {MAX_TAG_VAL_LENGTH} characters",
```
Comment (harupy) [inline tests/store/tracking/test_sqlalchemy_store.py:1311]: can we revert unrelated change?

File: tests/store/tracking/test_sqlalchemy_store.py (hunk 1 above)
Comment (harupy) [reply tests/store/tracking/test_sqlalchemy_store.py:495]: Can we insert a comment?
